dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -8370,9 +8537,20 @@ int cs_get_derived_plan(derived_handler* handler, THD* thd, SCSEP& csep, gp_walk
     return 0;
 }
 
-int cs_get_select_plan(select_handler* handler, THD* thd, SCSEP& csep, gp_walk_info& gwi)
+int cs_get_select_plan(ha_columnstore_select_handler* handler, THD* thd, SCSEP& csep, gp_walk_info& gwi)
 {
-    SELECT_LEX select_lex = *handler->select;
+    SELECT_LEX& select_lex = *handler->select;
```

Comment:
Why do you need this?
It looks more appropriate to work with a copy or make it a const ref.

Reply:
I am not sure about this. I don't see definitions of the copy constructor for `SELECT_LEX` so I am not sure what the behaviour of using the compiler generated copy ctor here would be. `getGroupPlan` (although obsolete) also uses a reference instead of a copy. Use of const ref generates a lot of compiler errors.